### PR TITLE
Address reviews in the proposal

### PR DESCRIPTION
**Pull request by nikhita**
cc @sttts

### CODE REVIEW

**Comment by sttts** [comment]
lgtm

File: contributors/design-proposals/customresources-validation.md [hunk 1]
```diff
@@ -306,6 +310,22 @@ The following example illustrates how a schema can be used in `CustomResourceDef
...
 ```
 
+### Error messages
+
+The following examples illustrate the type of validation errors generated by using the go-openapi validate library.
+
+1. `data.foo in body should be at least 4 chars long`
+2. `data.foo in body should be greater than or equal to 10`
+3. `data.foo in body should be one of [bar baz]`
+4. `data.foo in body must be of type integer: "string"`
```
Comment (nikhita) [inline contributors/design-proposals/customresources-validation.md:323]: Is this ok though? This occurs when I input a string instead of an integer.

Comment (nikhita) [reply contributors/design-proposals/customresources-validation.md:323]: I mean, should we have better error messages for this condition?

Comment (sttts) [reply contributors/design-proposals/customresources-validation.md:323]: sounds like a bug

Comment (sttts) [reply contributors/design-proposals/customresources-validation.md:323]: or is "string" the description?

Comment (nikhita) [reply contributors/design-proposals/customresources-validation.md:323]: No, string is the "type". It does not use the description field while generating errors. :confused:

File: contributors/design-proposals/customresources-validation.md [hunk 2]
```diff
@@ -368,7 +393,7 @@ Then we can validate the status against the sub-schema easily. Hence, this propo
 
 Apiextensions-apiserver uses the normal REST endpoint implementation and only customizes the registry and the codecs. The admission plugins are inherited from the kube-apiserver (when running inside of it via apiserver delegation) and therefore they are supposed to apply to CRs as well.
 
-The webhook admission prototyped at https://github.com/kubernetes/kubernetes/pull/46316 is expected to work with CRs out of the box. Hence, for more advanced validation the webhook admission is an option as well (when it is merged).
+It is [verified](https://github.com/kubernetes/kubernetes/pull/47252) that CRDs work well with initializers. Thus webhook admission prototyped at https://github.com/kubernetes/kubernetes/pull/46316 is also expected to work with CRs out of the box. Hence, for more advanced validation webhook admission is an option as well (when it is merged).
```
Comment (nikhita) [inline contributors/design-proposals/customresources-validation.md]: PTAL if we this is fine/anything else needs to be added.

Comment (sttts) [reply contributors/design-proposals/customresources-validation.md]: not sure the `Thus` is right. Just say "It is also expected...."

Comment (sttts) [reply contributors/design-proposals/customresources-validation.md]: Here is a link: https://github.com/kubernetes/kubernetes/pull/46316

File: contributors/design-proposals/customresources-validation.md [hunk 3]
```diff
@@ -44,6 +46,7 @@ Currently we do not provide validation for CustomResources (CR), i.e. the CR pay
 2. To keep open the door to add other validation mechanisms later.<sup id="f2">[2](#footnote2)</sup>
 3. To allow server-side validation.
 4. To be able to integrate into the existing client-side validation of kubectl.
+5. To be able to define defaults in the specification (at least in a follow-up of basic validation support).
```
Comment (sttts) [inline contributors/design-proposals/customresources-validation.md]: of => after

**Pull request closed without merging**